docs/book/src/tasks/using-kustomize.md

```diff
@@ -0,0 +1,154 @@
+# Using Kustomize with Workload Cluster Manifests
+
+Although the `clusterctl config cluster` command exposes a number of different configuration values 
+for customizing workload cluster YAML manifests, some users may need additional flexibility above 
+and beyond what `clusterctl config cluster` or the example "flavor" templates that some CAPI providers 
+supply (as an example, see [these flavor templates](https://github.com/kubernetes-sigs/cluster-api-provider-azure/tree/master/templates/flavors) 
+for the Cluster API Provider for Azure). In the future, a [templating solution](https://github.com/kubernetes-sigs/cluster-api/issues/3252) 
+may be integrated into `clusterctl` to help address this need, but in the meantime users can use 
+`kustomize` as a solution to this need.
+
+This document provides a few examples of using `kustomize` with Cluster API. All of these examples 
+assume that the "base" (unmodified) Cluster API configuration (perhaps generated using 
+`clusterctl config cluster`) is in a directory whose relative path is `../../base`.
+
+## Example: Using Kustomize to Specify Custom Images
+
+Users can use `kustomize` to specify custom OS images for Cluster API nodes. Using the Cluster API 
+Provider for AWS (CAPA) as an example, the following `kustomization.yaml` would leverage a JSON 6902 patch 
+to modify the AMI for nodes in a workload cluster:
+
+```yaml
+---
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+resources:
+  - ../../base
+patchesJson6902:
+  - path: custom-ami.json
+    target:
+      group: infrastructure.cluster.x-k8s.io
+      kind: AWSMachineTemplate
+      name: ".*"
+      version: v1alpha3
+```
+
+The referenced JSON 6902 patch looks like this:
+
+```json
+[
+    { "op": "add", "path": "/spec/template/spec/ami", "value": "ami-042db61632f72f145"}
+]
+```
+
+This configuration assumes that the workload cluster _only_ uses MachineDeployments. Since 
+MachineDeployments and the KubeadmControlPlane both leverage AWSMachineTemplates, this `kustomize` 
+configuration would catch all nodes in the workload cluster.
+
+## Example: Adding a MachineHealthCheck for a Workload Cluster
+
+Users could also use `kustomize` to combine additional resources, like a MachineHealthCheck (MHC), with the 
+base Cluster API manifest. In an overlay, specify the following in `kustomization.yaml`:
```

Comment:
Also in this case, having the filesystem tree could really help

Reply:
See changes in 0f98ee3aa.